### PR TITLE
{executorch][llama] support mqa

### PR DESCRIPTION
**Pull request by kimishpatel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #3080

This diff adds support for multi query attention for sdpa with kv cache

Differential Revision: [D56228316](https://our.internmc.facebook.com/intern/diff/D56228316/)

### CODE REVIEW

**Review by mikekgfb — APPROVED** [review]
Please ensure all tests pass before landing!

File: examples/models/llama2/custom_ops/TARGETS [hunk 1]
```diff
@@ -1,8 +1,22 @@
 # Any targets that should be shared between fbcode and xplat must be defined in
 # targets.bzl. This file can contain fbcode-only targets.
 
+load("@fbcode_macros//build_defs:python_unittest.bzl", "python_unittest")
```
Comment (larryliu0820) [inline examples/models/llama2/custom_ops/TARGETS:4]: Use `runtime`? You can do

```
runtime.python_unittest(
...
)
```

Comment (kimishpatel) [reply examples/models/llama2/custom_ops/TARGETS:4]: oh this was due to arc lint. let me fix.

**Comment by kimishpatel** [comment]
bunch of pre-existing failures with re2.h that @larryliu0820  is fixing

**Pull request closed without merging**